ocaml/xapi/xapi_vgpu_type.ml

```diff
@@ -64,13 +65,14 @@ module Identifier = struct
       match id with
       | Passthrough -> "passthrough"
       | Nvidia nvidia_id ->
-        Printf.sprintf "nvidia,%04x,%s,%04x,%04x"
+        Printf.sprintf "nvidia,%04x,%s,%04x,%04x,%s"
```

Comment:
The existing parameters are treated as hexadecimal numbers. Why are we using a string here for the new parameter?

Reply:
Hi, @lindig 
Actually type_id are not hexadecimal numbers.
See the `vgpuConfig.xml` file:
```
  <vgpuType id="11" name="GRID M60-0B" class="NVS">
    <devId vendorId="0x10de" deviceId="0x13F2" subsystemVendorId="0x10de" subsystemId="0x1176"></devId>
    <framebuffer>0x1A000000</framebuffer>
    <numHeads>2</numHeads>
    <display width="2560" height="1600"></display>
    <mappableVideoSize>0x400000</mappableVideoSize>
    <fbReservation>0x6000000</fbReservation>
    <frlConfig>0x2d</frlConfig>
    <cudaEnabled>0x0</cudaEnabled>
    <multiVgpuSupported>0x0</multiVgpuSupported>
    <encoderCapacity>0x64</encoderCapacity>
    <bar1Length>0x100</bar1Length>
    <frame_rate_limiter>1</frame_rate_limiter>
    <license>GRID-Virtual-PC,2.0;GRID-Virtual-WS,2.0;Quadro-Virtual-DWS,5.0;GRID-Virtual-WS-Ext,2.0</license>
```
Other number fields like deviceID are defined as hexadecimal numbers and start with 0x, but `vgpuType id ` is not treated as number here, I think string can be flexbility in the future?
Thanks.

Reply:
ok. It's not obvious what value ID can take, so let's keep the string.

Reply:
Hello, I am a bit rusty on how vGPU works now, but the format of the VGPU_type IDs is described [here](http://xapi-project.github.io/xapi/futures/vgpu-type-identifiers.html).

If you need to change the format of the identifier for existing VGPU_types you'll probably want to bump `VGPU_type.Identifier.version` and implement some upgrade logic. The identifier string is used on import as a primary key for identifying VGPU_types, as identical VGPU_types will have different UUIDs across different pools.

Reply:
:) Many thanks for your useful comment. 
- Yes, I will bump up the version.
- This PR is just some fundamental changes, and the more changes like upgrade logic will be added soon. Thanks for your remind, that's helpful.

Reply:
Is it really needed to extend the identifier string? Is the existing one not good enough anymore as a primary key? It seems to me that the type ID is redundant information, and we might as well leave it out so that we don't need to worry about the upgrade/import cases?

Reply:
Yes, now I will make changes to internal_config and delete the type_id from the identifier.